### ISSUE TITLE
Storybook failes to start when there is no node_modules folder

### ISSUE THREAD

**Issue by wintercounter** [storybookjs/storybook#8863]
As the title says.

The issue is with this line: `_path.default.join(cacheDir, 'records.json'),`

`cacheDir` will be undefined by `find-cache-path` if no node_modules exist.

It's just a minor bug as usually you tend to have node_modules.

**To Reproduce**
1. Go to your project root
2. Make sure you have no dependencies / delete node_modules
3. Run storybook

**Expected behavior**
Works :)

**Comment by ndelangen** [comment]
Someone installs storybook...

This adds dependencies...

So the situation where the user has no dependencies and no `node_modules` should never happen.

Or is there some way of adding storybook without adding dependencies/`node_modules`?

I think the expected behaviour is the error actually.

If you run `yarn install` then try to run storybook, that should resolve the error, right?

**Comment by wintercounter** [comment]
I'm having storybook globally installed (it actually comes from another global tool).

I found out this on a project with no dependencies because we're just displaying SVG icons in Storybook. We have one story file with `require.context` to all SVGs and using the mentioned global tool we generate Storybook to browse them. This project had a single dependency before, now we removed it, that's how we found out that it doesn't work. Creating an empty `node_modules` folder makes it work just fine.

**Comment by ndelangen** [comment]
I'm open to accepting a PR to fix this, but we don't want to actively support installing storybook globally. We don't document this is possible anywhere.

I suspect that:

> cacheDir will be undefined by find-cache-path if no node_modules exist.

Is just the tip of the iceberg

**Comment by JRGranell** [comment]
We've just run into this issue as well. We have a multi-package repo using lerna which "hoists" shared packages to repo root. This means that our package may not have any `node_modules` and storybook errors with the following message and cannot start.

```
Broken build, fix the error below.
WARN You may need to refresh the browser.
ERR! TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type undefined
ERR!     at validateString (internal/validators.js:107:11)
ERR!     at Object.join (path.js:1037:7)
ERR!     at _default (.../node_modules/@storybook/core/dist/server/manager/manager-webpack.config.js:144:32)
ERR!     at Object.managerWebpack(.../node_modules/@storybook/core/dist/server/manager/manager-preset.js:39:38)
ERR!     at ...node_modules/@storybook/core/dist/server/presets.js:82:72
```

Fixed by making an empty `node_modules` within each package - although this is currently  manual.

**Comment by ndelangen** [comment]
It seems there's a PR fixing this in find-cache-path:
https://github.com/avajs/find-cache-dir/pull/20

**Comment by pixelmanya** [comment]
Seems like this issue has been fixed in [v3.2.0](https://github.com/avajs/find-cache-dir/releases/tag/v3.2.0)
